lib/modulesync.rb

```diff
@@ -104,38 +110,31 @@ def self.manage_file(filename, settings, options)
     end
   end
 
-  def self.manage_module(puppet_module, module_files, module_options, defaults, options)
-    default_namespace = options[:namespace]
-    if module_options.is_a?(Hash) && module_options.key?(:namespace)
-      default_namespace = module_options[:namespace]
-    end
-    namespace, module_name = module_name(puppet_module, default_namespace)
-    git_repo = File.join(namespace, module_name)
-    unless options[:offline]
-      Git.pull(options[:git_base], git_repo, options[:branch], options[:project_root], module_options || {})
-    end
+  def self.manage_module(puppet_module, module_files, defaults)
+    Git.pull(puppet_module, options[:branch]) unless options[:offline]
 
-    module_configs = Util.parse_config(module_file(options[:project_root], namespace, module_name, MODULE_CONF_FILE))
+    module_configs = Util.parse_config(module_file(puppet_module, MODULE_CONF_FILE))
     settings = Settings.new(defaults[GLOBAL_DEFAULTS_KEY] || {},
                             defaults,
                             module_configs[GLOBAL_DEFAULTS_KEY] || {},
                             module_configs,
-                            :puppet_module => module_name,
+                            :puppet_module => puppet_module.repository_name,
                             :git_base => options[:git_base],
-                            :namespace => namespace)
+                            :namespace => puppet_module.repository_namespace)
+
     settings.unmanaged_files(module_files).each do |filename|
-      $stdout.puts "Not managing #{filename} in #{module_name}"
+      $stdout.puts "Not managing '#{filename}' in '#{puppet_module.given_name}'"
```

Comment:
Should the module have a `.to_s` which is `given_name`?

Reply:
ATM, not sure it's a good idea: `rubocop` will warn about usage of `to_s` but only mentionning `puppet_module` in string expansion can be confusing.
When refactoring will be done, I can apply your suggestion if you thing its relevant.